### ISSUE TITLE
Why does this angular formControl trigger @typescript-eslint/unbound-method

### ISSUE THREAD

**Issue by YoshiDatoutou** [typescript-eslint/typescript-eslint#1929]
{
  "rules": {
    "@typescript-eslint/unbound-method": "error"
  }
}

```ts
form: FormGroup = new FormGroup({
    assetId: new FormControl(null),
    name: new FormControl('', Validators.required),
    address: new FormControl(''),
    usDotNumber: new FormControl(''),
    phoneNumber: new FormControl(''),
    description: new FormControl(''),
  });
```

**What did you expect to happen?**
no eslint error, this is the format from angular example https://angular.io/guide/form-validation#built-in-validators
**What actually happened?**
error from eslint
![unbound error](https://user-images.githubusercontent.com/17058952/80153749-5c0c4100-857b-11ea-9aae-d72fb5fa01f5.PNG)

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`      | ^2.28.0 |
| `@typescript-eslint/parser`             | ^2.28.0 |
| `@typescript-eslint/typescript-estree`  | `X.Y.Z` |
| `@typescript-eslint/experimental-utils` | `X.Y.Z` |
| `TypeScript`                            | 3.5.3 |
| `node`                                  | 12.13.1 |
| `npm`                                   | 6.13.6 |

**Comment by bradzacher** [comment]
Please make sure you use the issue templates when you're filing an issue.
We use them to help keep things organised and ensure you give us information up front.

----

`Validators.requried` is a method on an object. It means it has an implicit `this` context.
You are using this method in a way such that it may become bound to a different `this` context, which could break the method.

Have a read of the rule's documentation: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/unbound-method.md

Note that typescript includes no information about things that are already bound, so if this is a false positive, use an eslint disable comment.

**Comment by YoshiDatoutou** [comment]
> Please make sure you use the issue templates when you're filing an issue.
> We use them to help keep things organised and ensure you give us information up front.
> 
> `Validators.requried` is a method on an object. It means it has an implicit `this` context.
> You are using this method in a way such that it may become bound to a different `this` context, which could break the method.
> 
> Have a read of the rule's documentation: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/unbound-method.md
> 
> Note that typescript includes no information about things that are already bound, so if this is a false positive, use an eslint disable comment.

I dont think it's possible to call "this" inside the new FormControl. I will stick with 
// eslint-disable-line @typescript-eslint/unbound-method
Thank you for the fast response.

**Comment by DutchKevv** [comment]
Thanks for this thread,

But I don't think adding eslint-disable-* lines is helping here, because tslint is deprecated, it needs a solution.

I have a LOT of forms, and placing a disable rule on every component doesn't give a warm feeling.. 
Specially cause I don't need it anywhere else :)

**Comment by bradzacher** [comment]
A few points as to why there's no solution to this from our side:

From the perspective of detecting if a function requires a specific `this` context:
- there is no way to statically analyse if a method actually requires a specific `this` context.
- the TypeScript typechecker also provides no information about this.
- there is no convention that exists in TypeScript code we could piggyback off of.

From the perspective of detecting if a function is already bound:
- there is no way to reliably statically analyse if a method is pre-bound to a `this` context.
- the TypeScript typechecker also provides no information about this.

From the perspective of detecting if a receiver can accept a function which needs a `this` context:
- there is no way to statically analyse if a receiver actually calls a given function
- the TypeScript typechecker also provides no information about this.

Within the context of this specific case:
[`Validators.required` is a static method](https://github.com/angular/angular/blob/master/packages/forms/src/validators.ts#L194-L196). HOWEVER, it's important to note that a static method still has access to a `this` context, but it is a different `this` context to what you're used to - it is a "static `this`".
So it being a static method is not a marker we can use to auto-ignore a method either (see #1639).

If you've got solutions that can be built into a lint rule - I would very happy to accept a PR!

Other solutions I can see:
- the angular library changes from a static method, to a static arrow function property.
- the angular library annotates its functions with something like `this: void` which we can build handling for in this rule
- you, the user, uses disable comments.
